pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+
+    function itScheduleGrantPermissionCorrectly(getSender: () => SignerWithAddress) {
+      it('schedules a grant permission', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        const { executed, data, where, executableAt } = await authorizer.getScheduledExecution(id);
+        expect(executed).to.be.false;
+        expect(data).to.be.equal(
+          authorizer.instance.interface.encodeFunctionData('grantPermission', [ACTION_1, user.address, WHERE_1])
+        );
+        expect(where).to.be.equal(authorizer.address);
+        expect(executableAt).to.equal((await currentTimestamp()).add(delay));
+      });
+
+      it('execution can be unprotected', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.false;
+      });
+
+      it('execution can be protected', async () => {
+        const executors = range(4).map(randomAddress);
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, executors, { from: getSender() });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.true;
+        await Promise.all(
+          executors.map(async (executor) => expect(await authorizer.isExecutor(id, executor)).to.be.true)
+        );
+      });
+
+      it('granter can cancel the execution', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+        expect(await authorizer.isCanceler(id, getSender())).to.be.true;
+
+        const receipt = await authorizer.cancel(id, { from: getSender() });
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionCancelled', { scheduledExecutionId: id });
+      });
+
+      it('can be executed after the expected delay', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        const receipt = await authorizer.execute(id);
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', { scheduledExecutionId: id });
+      });
+
+      it('grants the permission when executed', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_1, user, WHERE_1)).to.be.equal(true);
+      });
+
+      it('does not grant any other permissions when executed', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
```

Comment:
I really like these simple tests for the `cancel` and `execute` flow, without going crazy (e.g. not testing what happens if you execute a cancelled action, since we already test that in `execute` tests), but showing a reasonable happy path for all features.